### ISSUE TITLE
<script/style> tags in dynamically loaded panels should be hoisted to actually run

### ISSUE THREAD

**Issue by ang-zeyu** [MarkBind/markbind#2300]
### Please confirm that you have searched existing issues in the repo

Yes, I have searched the existing issues

### Any related issues?

#1565, #2130

### Tell us about your environment

Anything

### MarkBind version

Anything

### Describe the bug and the steps to reproduce it

This issue surfaced in implementing SSR #1565 and again in https://github.com/MarkBind/markbind/pull/2130#issuecomment-1501134894.

For `panel src=...`, `<script/style>` tags are not being hoisted out of the vue template during template compilation resulting in loss of commonmark functionality.

This is already being done for the main template run through SSR as part of #1565. Externals were specifically excluded from the hoisting process as it was out of scope. https://github.com/MarkBind/markbind/blob/master/packages/core/src/html/scriptAndStyleTagProcessor.ts#L19

### Expected behavior

We should probably be extending the fix to `External`s as well.

i.e.,
- `script` tags should be temporarily extracted out of the template before compilation in `Retriever.vue` then reinserted for execution later.
- The same for `style` tags.

### Anything else?

Hoisting as a solution to Vue disallowing `script/style`s was adapted from Vuepress.

Once this is done, we can look into a more proper fix that retains the `script` element placement instead of hoisting it downward.

Use case: inline scripts that reference `document.currentScript` to return the current `script` element may be functionally affected.

There should be no issue for `style`.

**Comment by yiwen101** [comment]
Will look into it.
